src/gt4py/next/iterator/transforms/transform_broadcast.py

```diff
@@ -0,0 +1,35 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+import dataclasses
+
+from gt4py.eve import NodeTranslator, PreserveLocationVisitor
+from gt4py.next.iterator import ir as itir
+from gt4py.next.iterator.ir_utils import (
+    common_pattern_matcher as cpm,
+    domain_utils,
+    ir_makers as im,
+)
+
+
+@dataclasses.dataclass
+class TransformBroadcast(PreserveLocationVisitor, NodeTranslator):
```

Comment:
This pass is trivial so a separate unit test file is too much I would say, how about a small docstring + doctest here.